### PR TITLE
Prevent use Pump Up in patches

### PR DESCRIPTION
**Pull request by ErdGinalD**
## What Does This PR Do
Slightly nerf Pump Up not allowing to work while used in patches, making most cheap and still very good drug without downsides to be absolutely best drug in the game. 

## Why It's Good For The Game
Rupara always on top of most unfunny mechanic abuses. Pump Up became most usable drug in the game, because someone invent to put this in patches. While other drugs have downside effects, which effectively prevent usage in patches, Pump Up doesnt have any of them, providing nearly-eternal antistun without consequenses. Instead of nerfing Pump Up, killing him entirely, i add minimal volume isage, so you can still use it manually, but not in patches.

## Images of changes
*screams of patсhes become useless*
image

## Testing
copypasta from stimulants, good work without bugs.

## Changelog
:cl:
tweak: Pump Up need at least 1u in body to work, preventing its use in patches.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 1]
```diff
@@ -259,13 +259,14 @@
...
 
 /datum/reagent/pump_up/on_mob_life(mob/living/M)
-	var/update_flags = STATUS_UPDATE_NONE
-	update_flags |= M.adjustStaminaLoss(-21) // one cycle to get out of stam crit ~2 second
-	M.AdjustParalysis(-2 SECONDS)
-	M.AdjustStunned(-2 SECONDS)
-	M.AdjustWeakened(-2 SECONDS)
-	M.AdjustKnockDown(-2 SECONDS)
-	return ..() | update_flags
+	if(volume > 0.6) // no patch use for ya
```
Comment (DGamerL) [inline code/modules/reagents/chemistry/reagents/drugs.dm:262]: Make this an early return

**Review by DGamerL — CHANGES_REQUESTED** [review]
This will make any amount of pump up under 0.6 not metabolize out

**Review by DGamerL — APPROVED** [review]
The code looks to work.

Not quite sure if restricting it like this is the best way, imo we could rather add a meaningful downside to it (and change it's 1u metabolize speed to be lower)

**Comment by ErdGinalD** [comment]
> The code looks to work.
> 
> Not quite sure if restricting it like this is the best way, imo we could rather add a meaningful downside to it (and change it's 1u metabolize speed to be lower)



> The code looks to work.
> 
> Not quite sure if restricting it like this is the best way, imo we could rather add a meaningful downside to it (and change it's 1u metabolize speed to be lower)



> The code looks to work.
> 
> Not quite sure if restricting it like this is the best way, imo we could rather add a meaningful downside to it (and change it's 1u metabolize speed to be lower)

no need in another crank but weaker, but use it in patches too silly

**Comment by farie82** [comment]
Hey there, sorry for the late response to the PR.
The PR will be closed since the balance team has voted against this PR with the following reasons:
- I have not seen this done on our codebase, and frankly, it seems weird to prevent it to be patch'd
- Don't like the idea of this either. The author said that this is a pretty big thing on RU-Para, but since it's not a problem here, I don't like the idea of changing it here.
- Why punish other forms of antistun that allow for more risky but creative builds that dont use adrenals? You still gotta actually apply the patch. Dont see why another server needs to affect our server.
- It's not being abused here so I dont think we should prevent it being in patches
- If anything you just need to fix the drug itself to provide a downside in patch form, not prevent it from being used in patches. Seems like using a sledgehammer to fix a minor issue.

**Pull request closed without merging**